syntax = "proto3";

package product_analysis;

option go_package = "github.com/yourusername/ecomate/proto;product_analysis";

service ProductAnalysisService {
  rpc AnalyzeProduct(AnalyzeProductRequest) returns (AnalyzeProductResponse);
  rpc CalculateCO2Impact(CalculateCO2Request) returns (CalculateCO2Response);
}

message AnalyzeProductRequest {
  repeated bytes images = 1;
  string title = 2;
  string user_provided_description = 3;
  string category = 4;
}

message AnalyzeProductResponse {
  string generated_description = 1;
  int32 suggested_price = 2;
  float estimated_weight_kg = 3;
  string manufacturer_country = 4;
  int32 estimated_manufacturing_year = 5;
  float co2_impact_kg = 6;
  bool is_inappropriate = 7;
  string inappropriate_reason = 8;
  repeated string detected_objects = 9;
}

message CalculateCO2Request {
  string category = 1;
  float weight_kg = 2;
  string manufacturer_country = 3;
  int32 manufacturing_year = 4;
}

message CalculateCO2Response {
  float buying_new_kg = 1;
  float buying_used_kg = 2;
  float saved_kg = 3;
}
